db/local_db/start_database.go

```diff
@@ -244,26 +252,131 @@ func startPostgresProcess(port int, listen StartListenType, invoker constants.In
 	}
 	err = runningInfo.Save()
 	if err != nil {
-		postgresCmd.Process.Kill()
+		killOnStartFail(postgresCmd)
 		return err
 	}
 
-	err = postgresCmd.Process.Release()
+	connection, err := createDbClient("postgres", constants.DatabaseSuperUser)
 	if err != nil {
-		postgresCmd.Process.Kill()
+		dumpDiagnosticData()
+		killOnStartFail(postgresCmd)
 		return err
 	}
+	connection.Close()
 
-	connection, err := createDbClient("postgres", constants.DatabaseSuperUser)
+	err = postgresCmd.Process.Release()
 	if err != nil {
-		postgresCmd.Process.Kill()
+		killOnStartFail(postgresCmd)
 		return err
 	}
-	connection.Close()
 
 	return nil
 }
 
+func killOnStartFail(cmd *exec.Cmd) {
+	// get the process
+	p, _ := psutils.NewProcess(int32(cmd.Process.Pid))
+	// kill it
+	doThreeStepPostgresExit(p)
+}
+
+func dumpDiagnosticData() {
+	dumpingDone := make(chan bool, 1)
+	sp := display.StartSpinnerAfterDelay("Fatal error occurred. Collecting diagnostic data", constants.SpinnerShowTimeout, dumpingDone)
+	dump := DiagnosticDumpData{}
+
+	wg := sync.WaitGroup{}
+
+	// collector is a wrapper over the actual collector functions and provides the necessary concurrency
+	// support
+	collector := func(fn func(d *DiagnosticDumpData), sink *DiagnosticDumpData, wg *sync.WaitGroup) {
```

Comment:
on second thoughts, lets declare this function at file scope, and call it runCollectionAsync 

and rename the functions collect*, i.e. `collectProcesses`.